src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+
+   // Getting -2LogL
+   const double my_likelihood = lilith_computelikelihood(lilithcalc2);
+
+   Py_Finalize();
+   return my_likelihood;
+}
+#endif
+
+void set_sm_settings_matching_bsm(
+   standard_model::Standard_model& sm,
+   Spectrum_generator_settings const& spectrum_generator_settings)
+{
+  sm.set_pole_mass_loop_order(static_cast<int>(spectrum_generator_settings.get(Spectrum_generator_settings::pole_mass_loop_order)));
+  sm.set_ewsb_loop_order(static_cast<int>(spectrum_generator_settings.get(Spectrum_generator_settings::ewsb_loop_order)));
+  sm.set_precision(spectrum_generator_settings.get(Spectrum_generator_settings::precision));
+  sm.set_threshold_corrections(spectrum_generator_settings.get_threshold_corrections());
+  sm.set_loop_corrections(spectrum_generator_settings.get_loop_corrections());
+  sm.set_loops(static_cast<int>(spectrum_generator_settings.get(Spectrum_generator_settings::beta_loop_order)));
+
+  Loop_corrections loop_corrections_;
+  loop_corrections_.higgs_at_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_2loop_correction_at_as);
+  loop_corrections_.higgs_ab_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_2loop_correction_ab_as);
+  loop_corrections_.higgs_at_at = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_2loop_correction_at_at);
+  loop_corrections_.higgs_atau_atau = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_2loop_correction_atau_atau);
+  loop_corrections_.top_qcd = spectrum_generator_settings.get(Spectrum_generator_settings::top_pole_qcd_corrections);
+  loop_corrections_.higgs_at_as_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_3loop_correction_at_as2);
+  loop_corrections_.higgs_ab_as_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_3loop_correction_ab_as2);
+  loop_corrections_.higgs_at_at_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_3loop_correction_at2_as);
+  loop_corrections_.higgs_at_at_at = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_3loop_correction_at3);
+  loop_corrections_.higgs_at_as_as_as = spectrum_generator_settings.get(Spectrum_generator_settings::higgs_4loop_correction_at_as3);
+  sm.set_loop_corrections(loop_corrections_);
+
+  sm.set_threshold_corrections(spectrum_generator_settings.get_threshold_corrections());
+}
+
+void set_sm_lambda_to_match_bsm_mh(standard_model::Standard_model& sm, double mass, std::string const& name) {
+
+      // set SM λ such that mhSM == mass
+      auto match_Higgs_mass = [&sm, mass](double x) {
+         sm.set_Lambdax(x);
+         sm.calculate_DRbar_masses(); // internaly calls solve_ewsb_tree_level()
+         sm.solve_ewsb();
+         sm.calculate_Mhh_pole();
+         return std::abs(sm.get_physical().Mhh - mass);
+      };
+
+      int status, iter = 0;
+      static constexpr int max_iter = 100;
+      const gsl_min_fminimizer_type *T;
```

Comment:
Could you please open another issue to refactor this C-style minimization using GSL? This workaround and the manual allocation and (not exception-safe) manual deallocation makes me nervous. :)

Reply:
Done (#641)